dev-tools/validation/check-authz-route-matrix.py

```
@@ -114,6 +176,629 @@ def validate_auth_path_vocabulary(document: dict[str, Any], errors: list[str]) -
+) -> None:
+    for index, route in enumerate(routes):
+        if not isinstance(route, dict) or "role_assurance" not in route:
+            continue
+        assurance = route.get("role_assurance")
+        if not isinstance(assurance, str) or assurance not in predicates:
+            errors.append(
+                f"routes[{index}] role_assurance must reference a declared predicate"
+            )
+
+
+def validate_generation_applicability(routes: list[Any], errors: list[str]) -> None:
```

Ruff (0.15.21)

[warning] Too many branches (19 > 12)

(PLR0912)